cmd/kubeadm/app/phases/certs/certlist.go

```diff
@@ -61,24 +62,53 @@ func (k *KubeadmCert) CreateFromCA(ic *kubeadmapi.InitConfiguration, caCert *x50
 	if err != nil {
 		return err
 	}
-	writeCertificateAuthorithyFilesIfNotExist(
+	writeCertificateFilesIfNotExist(
 		ic.CertificatesDir,
 		k.BaseName,
+		caCert,
 		cert,
 		key,
 	)
 
 	return nil
 }
 
+// CreateAsCA creates a certificate authority, writing the files to disk and also returning the created CA so it can be used to sign child certs.
+func (k *KubeadmCert) CreateAsCA(ic *kubeadmapi.InitConfiguration) (*x509.Certificate, *rsa.PrivateKey, error) {
+	cfg, err := k.GetConfig(ic)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't create %q CA certificate: %v", k.Name, err)
```

Comment:
These errors are all the same but they should be specific to the failure that occurred. I'd expect the message "couldn't create..." from whatever is calling `CreateAsCA` and I'd expect the error here to be `"failed to get config for kubeadm cert %q: %v", k.Name, err`